### ISSUE TITLE
Style/ArrayIntersect autocorrects `none? { |s| str.include?(s) }` into a TypeError

### ISSUE THREAD

**Issue by modosc** [rubocop/rubocop#15438]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Style/ArrayIntersect

### Steps to reproduce the problem

```ruby
# frozen_string_literal: true

IGNORED_SUBSTRINGS = ["index", "idx", "pkey"].freeze

def keep?(relname)
  IGNORED_SUBSTRINGS.none? { |substring| relname.include?(substring) }
end

keep?("users_pkey") # => false
```

```
repro.rb:6:3: C: [Correctable] Style/ArrayIntersect: Use !IGNORED_SUBSTRINGS.intersect?(relname)
instead of IGNORED_SUBSTRINGS.none? { |substring| relname.include?(substring) }.
```

and after `rubocop -A` the code raises:

```
TypeError: no implicit conversion of String into Array (Array#intersect?)
```

i believe this is different from #15437 because `rubocop` is falsely assuming that `relname` is an `Array` when it's not. 

### Expected behavior

`array.none? { |e| receiver.include?(e) }` should pass when `receiver` isn't known to be an array.

### Actual behavior

```
TypeError: no implicit conversion of String into Array (Array#intersect?)
```

### RuboCop configuration

```yaml
AllCops:
  NewCops: enable
```

### RuboCop version

```shell
1.88.2 (using Parser 3.3.11.1, Prism 1.9.0, parser_prism, rubocop-ast 1.50.0, analyzing as Ruby 4.0, running on ruby 4.0.5) [arm64-darwin25]
  - rubocop-rspec 3.10.2
  - rubocop-rails 2.35.5
  - rubocop-performance 1.26.1
  - rubocop-factory_bot 2.28.0
  - rubocop-rspec_rails 2.32.0
```
